src/Sulu/Bundle/ContentBundle/Preview/PageRouteDefaultsProvider.php

```diff
@@ -88,7 +88,10 @@ public function isPublished($entityClass, $id, $locale)
      */
     public function supports($entityClass)
     {
-        return HomeDocument::class === $entityClass || PageDocument::class === $entityClass;
+        return HomeDocument::class === $entityClass
+            || PageDocument::class === $entityClass
+            || is_subclass_of($entityClass, HomeDocument::class)
+            || is_subclass_of($entityClass, PageDocument::class);
```

Comment:
Why not just `$entityClass instanceof HomeDocument || $entity instanceof PageDocument`?